microsetta_public_api/api/taxonomy.py

```diff
@@ -45,6 +46,59 @@ def summarize_group(body, resource):
     return _summarize_group(sample_ids, resource, taxonomy_repo)
 
 
+LEVEL_MAP = {
+    'domain': 0,
+    'phylum': 1,
+    'class': 2,
+    'order': 3,
+    'family': 4,
+    'genus': 5,
+    'species': 6
+}
+
+
```

Comment:
It would be a lot cleaner if we add a function like this:

```suggestion
def _check_resource_and_missing_ids_alt(taxonomy_repo, sample_ids, resource):
    available_resources = taxonomy_repo.resources()
    type_ = 'resource'
    validate_resource_alt(available_resources, resource, type_)
    missing_ids = [id_ for id_ in sample_ids if
                   not taxonomy_repo.exists(id_, resource)]
    check_missing_ids_alt(missing_ids, resource, type_)

```

We would have to import `validate_resource_alt` and `check_missing_ids_alt` from `microsetta_public_api.utils._utils`

Reply:
Then the 

```python
error_response = fn(...)
if error_response:
    return error_response
```

lines below just become

```python
fn(...)
```

Reply:
All the `_alt` stuff is confusing to me. To be honest, I'm not sure how the suggestion improves clarity here. It looks like it's more complex than what's currently there?

Reply:
Maybe not the best naming scheme, but still:
https://github.com/biocore/microsetta-public-api/pull/73/files#r540501770
https://github.com/biocore/microsetta-public-api/issues/77